content/graph/Reroot.h

```diff
@@ -0,0 +1,101 @@
+/**
+ * Author: Pietro Palombini
+ * Date: 2024-07-06
+ * License: CC0
+ * Source: https://codeforces.com/blog/entry/124286
+ * Description: Usage: define \texttt{Data}, \texttt{finalize},
+ * \texttt{acc} according to the definitions below, adding any
+ * necessary fields to \texttt{Reroot}. Then instantiate, add
+ * the data, and call \texttt{reroot()}.
+ *
+ * \texttt{finalize} is applied before returning
+ * answer for \texttt{p} if \texttt{g[p][ei]} was its
+ * parent; usually identity function.
+ *
+ * Let accumulated$(p, S) = a$ such that \texttt{finalize(a,
+ * ...)} is the answer for $p$ if $S$ were its only children.
+ * \texttt{acc} should, given accumulated$(p, S)$ and the
+ * answer for \texttt{g[p][ei]}, compute
+ * accumulated$(p,S\cup\{\texttt{g[p][ei]}\})$ in
+ * \texttt{p\_ans}.
+ *
+ * \texttt{neuts[p]} should be accumulated$(p, \varnothing)$.
+ *
+ * The answers are \texttt{root\_dp[v]}, the answer for the
+ * whole tree with \texttt{v} as root, \texttt{fdp[v][ei]}, the
+ * answer for \texttt{g[v][ei]} if \texttt{v} is the root, and
+ * \texttt{bdp[v][ei]}, the answer for \texttt{v} if
+ * \texttt{g[v][ei]} is the root.
+ *
+ * Equivalent to the following code, but for all roots:
+ *
+ * \texttt{Data dfs(ll u, ll p) \{}
+ *
+ * \ \ \texttt{Data res = neuts[u];}
+ *
+ * \ \ \texttt{fore(ei, 0, SZ(g[u])) if (g[u][ei] != p)}
+ *
+ * \ \ \ \ \texttt{acc(res, dfs(g[u][ei], u), u, ei);}
+ *
+ * \ \ \texttt{ll pid = find(ALL(g[u]), p) - begin(g[u]);}
+ *
+ * \ \ \texttt{return dp[u] = finalize(res, u, pid);}
+ *
+ * \texttt{\}}
+ * Time: Fast O(n \log n) assuming O(1) operations.
+ * Status: stress-tested
+ */
+
+#pragma once
+
+struct Data {};
+typedef vector<Data> vd;
+
+struct Reroot {
+	ll n;
+	vector<vi>& g;
+	vd& neuts;
+
+	Data finalize(const Data& a, ll p, ll ei) {
+		return a;
+	}
+	void acc(Data& p_ans, const Data& child_ans, ll p, ll ei) {
+		p_ans = Data{};
```

Comment:
```suggestion
		p_ans = {};
```
¿Algo así no anda? (En los otros archivos también aplica esta sugerencia)

Reply:
Sí anda creo. Es por claridad nomás. Eso no se copia realmente sino que lo vas a definir según el problema.